apps/modernization-ui/src/apps/page-builder/pages/AddNewPage/AddNewPageFields.tsx

```diff
@@ -50,12 +51,16 @@ export const AddNewPageFields = (props: AddNewPageFieldProps) => {
             <Controller
                 control={form.control}
                 name="name"
-                rules={{ required: { value: true, message: 'Name is required.' } }}
+                rules={{
+                    required: { value: true, message: 'Name is required.' },
+                    ...validPageNameRule
+                }}
                 render={({ field: { onChange, value }, fieldState: { error } }) => (
                     <Input
                         onChange={onChange}
                         defaultValue={value}
                         label="Page name"
+                        className="pageName"
```

Comment:
Issue: You will need to pass the `onBlur` callback into the Input so that the validation will trigger when the user tabs away from the input.

```typescript
 render={({ field: { onChange, onBlur, value }, fieldState: { error } }) => (
                    <Input
                        onBlur={onBlur}
                        onChange={onChange}
                        defaultValue={value}


```